tests/include_doctest.h

```diff
@@ -0,0 +1,7 @@
+#pragma once
+#define DOCTEST_CONFIG_IMPLEMENT
+#if __has_include(<doctest.h>)
+#include <doctest.h>
+#else
+#include "doctest.h"
+#endif
```

Comment:
All of this is not really necessary. You can just `#include <doctest.h>` and add the define(s) you want for doctest to one of the test source files:

```cmake
set_property(SOURCE tests/tests.cpp PROPERTY COMPILE_DEFINITIONS DOCTEST_CONFIG_IMPLEMENT)
```

Reply:
Nothing to add to that.